### PR TITLE
Fix/pipeline gas import tests

### PR DESCRIPTION
**Pull request by pworschischek-aggmag**
## Changes proposed in this Pull Request
Fixes tests on the main branch. 

## Checklist

**Required:**
- [x] Changes are tested locally and behave as expected.
- [x] Code and workflow changes are documented.
- [ ] All Sourcery Bot review suggestions have been implemented or discarded with an explanation.
- [ ] All github actions succeed.

## Summary by Sourcery

Update gas pipeline import blocking logic and associated tests to rely on aggregated statistics and actual pipeline presence rather than configuration-only assumptions.

Enhancements:
- Determine active countries for Russian gas import blocking based on existing pipeline gas generators instead of configured countries, and fail fast when expected generators are missing.

Tests:
- Refactor gas storage clustering and Russian gas import blocking tests to use shared clustering helpers and aggregated capacity/supply statistics, including explicit handling of the Bulgarian TANAP capacity edge case.

### CODE REVIEW

File: test/test_mods/network/test_gas.py [hunk 1]
```diff
@@ -29,10 +30,7 @@ def test_gas_storage_update(nc, project_root, is_testrun):
     expected = expected[expected > 0]
     # aggregate update values depending on custom clustering
     clustering = require_config(nc, "mods", "modify_nuts3_shapes")
-    if clustering.startswith("AT10"):
-        expected = expected.groupby(expected.index.map(_map_at_nuts3_to_nuts2)).sum()
-    if clustering.endswith("DE5"):
-        expected = expected.groupby(expected.index.map(_map_de_nuts1_to_de5)).sum()
+    expected = combine_regions_by_clustering(expected, clustering)
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply test/test_mods/network/test_gas.py:33]: overkill for an integration test and low complexity of helper function. Besides, the helper has its unit tests.

File: test/test_mods/network/test_gas.py [hunk 2]
```diff
@@ -90,40 +88,42 @@ def test_block_russian_gas_imports(nc, block_name):
...
-        for cc in countries:
-            if cc == "BG":
-                # supply has no fixed value for BG, since TANAP capacity is available
-                continue
-            assert (supply_annual == 0).all()(
-                f"Expected zero pipeline gas supply in {cc} for {year}, is {supply_annual}."
-            )
+        drop_zero=False,
+    ).pipe(filter_by, country=countries_wo_bg)
+    assert pipeline_supply.sum() == 0, (
```
[the remark this answers is not in the document]
Comment (pworschischek-aggmag) [reply test/test_mods/network/test_gas.py:127]: Its OK for 0.0 == 0 comparisons
